vcenter_operator/configurator.py

```diff
@@ -296,36 +296,72 @@ def setup_os_session(self):
         self.os_session = Session(auth=auth)
 
     def _poll_nova_cells(self):
+        """Fetch information about Nova's cells"""
         namespace_nova = self.global_options['namespace']
-
         label_selector = 'system=openstack,component=nova,type=nova-cell'
+
+        # We either read a list of cell names into dict keys or a dict of cell
+        # name -> connection info into the global option "cells". Templates
+        # should be changed to use only the names, we keep the reading of
+        # Secrets for not-yet-updated templates.
+        configmap_return = self._poll_nova_cells_from_configmap(namespace_nova, label_selector)
+
+        secrets_return = self._poll_nova_cells_from_secrets(namespace_nova, label_selector)
+
+        return configmap_return or secrets_return
```

Comment:
Is there a reason to read them both in contrast to, say:
```python

if configmap := self._poll_nova_cells_from_configmap(namespace_nova, label_selector):
  return configmap

return self._poll_nova_cells_from_secrets(namespace_nova, label_selector)
```

Reply:
Yes, I first had it that way, but if we still have the detailed information in Nova in the Secrets, we probably still rely on them in the templates. Therefore, depending on in which order we roll out stuff, it might make sense to read both until all templates are updated everywhere.